tests/components/blink/test_config_flow.py

```diff
@@ -0,0 +1,204 @@
+            key_required=False,
+            login_response={},
+        ),
+    ), patch(
+        "homeassistant.components.blink.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.blink.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"username": "blink@example.com", "password": "example"},
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "blink"
+    assert result2["result"].unique_id == "blink@example.com"
+    assert result2["data"] == {
+        "username": "blink@example.com",
+        "password": "example",
+        "login_response": {},
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import(hass):
+    """Test we import the config."""
+    with patch(
+        "homeassistant.components.blink.config_flow.Blink",
+        return_value=Mock(
+            get_auth_token=Mock(return_value=True),
+            key_required=False,
+            login_response={},
+        ),
+    ), patch(
+        "homeassistant.components.blink.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={
+                "username": "blink@example.com",
+                "password": "example",
+                "scan_interval": 10,
+            },
+        )
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == "blink"
+    assert result["result"].unique_id == "blink@example.com"
+    assert result["data"] == {
+        "username": "blink@example.com",
+        "password": "example",
+        "scan_interval": 10,
+        "login_response": {},
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_2fa(hass):
+    """Test we get the 2fa form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    mock_blink = Mock(
+        get_auth_token=Mock(return_value=True),
+        key_required=True,
+        login_response={},
+        login_handler=Mock(send_auth_key=Mock(return_value=True)),
+    )
+
+    with patch(
+        "homeassistant.components.blink.config_flow.Blink", return_value=mock_blink
+    ), patch(
+        "homeassistant.components.blink.async_setup", return_value=True
+    ) as mock_setup:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"username": "blink@example.com", "password": "example"}
+        )
+
+    assert result2["type"] == "form"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5705368)